datastore-design/go-grpc-demo/storagebackend/timescaledb/timescaledb.go

```diff
@@ -21,14 +22,22 @@ func (sbe *TimescaleDB) Description() string {
 
 // openDB opens database identified by host/port/user/password/dbname.
 // Returns (DB, nil) upon success, otherwise (..., error).
-func openDB(host, port, user, password, dbname string) (*sql.DB, error) {
-	connInfo := fmt.Sprintf(
+func openDB(host, port, user, password, dbname string, maxConns int32) (*pgxpool.Pool, error) {
+	connString := fmt.Sprintf(
 		"host=%s port=%s user=%s password=%s dbname=%s sslmode=disable",
 		host, port, user, password, dbname)
 
-	db, err := sql.Open("postgres", connInfo)
+	connConf, err := pgxpool.ParseConfig(connString)
```

Comment:
Is the pool still needed, as the problem was caused by not having the defer?

They way gRPC works, I think it should not make to many connections to the DB, even under high load.
Can you check the "KNMI data load" without the pool. If it works, I would rather go back to the simple "no pool" approach, until it is actually needed.